opium_kernel/core/rock.ml

```diff
@@ -153,6 +205,16 @@ module Response = struct
         ])
   ;;
 
+  let string_of_t t =
```

Comment:
Is this done just to avoid s-exp formatting? I had added pretty printers for all the with this usecase in mind. Getting a string from the formatter can be done via `Format.asprintf`

Reply:
I added these functions to improve the formatting. I find the s-exp not very readable.

Here's an extract of the logs with the s-exp:
```
middleware_debugger.exe: [DEBUG] ((version (major 1) (minor 1)) (status 500)
                                  (reason ()) (headers ())
                                  (body
                                   ((length (6810))
                                    (content
                                      "\
                                     \n<!doctype html>\
                                     \n<html lang=\"en\">\
                                     \n\
                                     \n<head>\
```

Compared to the logs with an HTTP format:

```
middleware_debugger.exe: [DEBUG] HTTP/1.1 500 



<!doctype html>
<html lang="en">

<head>
```

Same goes for the debugger.

<img width="1751" alt="Screenshot 2020-06-02 at 11 29 34" src="https://user-images.githubusercontent.com/6162008/83504434-cb534a00-a4c4-11ea-8ddd-c54808636519.png">

Compared with:

<img width="1749" alt="Screenshot 2020-06-02 at 11 29 14" src="https://user-images.githubusercontent.com/6162008/83504306-a3fc7d00-a4c4-11ea-9452-58268a855720.png">


But I agree about the flexibility. I added a formatted called `pp_http` in both Request and Response and use them in the logger and debugger.